src/biconnectivity/articulation.jl

```diff
@@ -78,3 +78,62 @@ function articulation(g::AbstractGraph)
     end
     return findall(state.articulation_points)
 end
+
+function iterative_articulation(g::AbstractGraph)
+    s = Array{NTuple{4,Int}}(undef,0)
+    articulation_points = falses(nv(g))
+    low = zeros(nv(g))
+    pre = zeros(nv(g))
+    children = 0
+    wi = 0
+    w = 0
```

Comment:
`w = zero(eltype(g))` or even better `w::eltype(g) = 0`. Otherwise there will be some type instability when you set `w = x[wi]` later.